src/lib/import-export/import/importers/importer.ts

```diff
@@ -14,35 +17,61 @@ export interface ImporterResult extends Omit< BackupContents, 'metaFile' > {
 }
 
 export interface Importer extends Partial< EventEmitter > {
-	import( rootPath: string ): Promise< ImporterResult >;
+	import( rootPath: string, siteId: string ): Promise< ImporterResult >;
 }
 
 export class DefaultImporter extends EventEmitter implements Importer {
 	constructor( protected backup: BackupContents ) {
 		super();
 	}
 
-	async import( rootPath: string ): Promise< ImporterResult > {
+	async import( rootPath: string, siteId: string ): Promise< ImporterResult > {
 		this.emit( ImportEvents.IMPORT_START );
 
-		await this.importDatabase();
-		await this.importWpContent( rootPath );
-		let meta: MetaFileData | undefined;
-		if ( this.backup.metaFile ) {
-			meta = await this.parseMetaFile();
+		try {
+			await this.importDatabase( rootPath, siteId );
+			await this.importWpContent( rootPath );
+			let meta: MetaFileData | undefined;
+			if ( this.backup.metaFile ) {
+				meta = await this.parseMetaFile();
+			}
+
+			this.emit( ImportEvents.IMPORT_COMPLETE );
+			return {
+				extractionDirectory: this.backup.extractionDirectory,
+				sqlFiles: this.backup.sqlFiles,
+				wpContent: this.backup.wpContent,
+				meta,
+			};
+		} catch ( error ) {
+			this.emit( ImportEvents.IMPORT_ERROR, error );
+			throw error;
 		}
-		this.emit( ImportEvents.IMPORT_COMPLETE );
-		return {
-			extractionDirectory: this.backup.extractionDirectory,
-			sqlFiles: this.backup.sqlFiles,
-			wpContent: this.backup.wpContent,
-			meta,
-		};
 	}
 
-	protected async importDatabase(): Promise< void > {
+	protected async importDatabase( rootPath: string, siteId: string ): Promise< void > {
+		if ( ! this.backup.sqlFiles.length ) {
+			return;
+		}
+
+		const server = SiteServer.get( siteId );
+		if ( ! server ) {
+			throw new Error( 'Site not found.' );
+		}
+
 		this.emit( ImportEvents.IMPORT_DATABASE_START );
-		// will implement in a different ticket
+		for ( const sqlFile of this.backup.sqlFiles ) {
```

Comment:
Should we sort the files by name ascending, or are they already sorted at this point?

Reply:
They will be sorted according to how they were added to the archive so that could theoretically be any order. It would not hurt to sort this list, just to be sure. Thanks for pointing that out!